pulp_python/app/tasks.py

```diff
@@ -62,6 +82,75 @@ def sync(importer_pk, repository_pk):
             changeset.apply_and_drain()
 
 
+@shared_task(base=UserFacingTask)
+def publish(publisher_pk, repository_pk):
+    """
+    Use provided publisher to create a Publication based on a RepositoryVersion.
+    Args:
+        publisher_pk (str): Use the publish settings provided by this publisher.
+        repository_pk (str): Create a Publication from the latest version of this Repository.
+    """
+    publisher = python_models.PythonPublisher.objects.get(pk=publisher_pk)
+    repository = models.Repository.objects.get(pk=repository_pk)
+    latest_version = models.RepositoryVersion.latest(repository)
+
+    log.info(_('Publishing: repository={repository}, version={version}, publisher={publisher}').format(
+        repository=repository.name,
+        version=latest_version.number,
+        publisher=publisher.name
+    ))
+
+    with WorkingDirectory():
+
+        os.mkdir('simple')
+        with models.Publication.create(latest_version, publisher) as publication:
+
+            packages_in_version = python_models.PythonPackageContent.objects.filter(
+                pk__in=publication.repository_version.content).order_by('-created')
+
+            aggregated_packages = itertools.groupby(packages_in_version, lambda x: x.name)
+            for (project_name, distributions) in aggregated_packages:
```

Comment:
s/distributions/packages/ 

Since you use packages elsewhere, it would be nice to stay consistent and specifically avoid the naming collision with pulpcore Distributions.